### PR TITLE
feat: use the code way to install core-dns

### PR DESCRIPTION
**Pull request by qiuwei68**
#### What type of PR is this?
/kind feature

#### What does this PR do?

use the code way to install core-dns
#### Which issue(s) does this PR fix?

Fixes #
#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: hack/globalnode_patch.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+# define the node name prefix
+node_prefix="test-ylc-m-"
```
Comment (duanmengkk) [inline hack/globalnode_patch.sh]:
```suggestion
node_prefix=""
```

File: hack/globalnode_patch.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+# Change the node label you want in the for loop, such as 01-09 in this case
+for ((i=1; i<=9; i++))
+do
+    # Use printf to format strings so that the number part is always two digits long
+    node_number=$(printf "%02d" $i)
+    node_name="$node_prefix$node_number"
+
+    # Update the status of the corresponding globalnode to "free" using the kubectl patch command.
+    kubectl patch globalnode $node_name --type=json -p '[{"op":"replace", "path":"/spec/state", "value":"free"}]'
+done
```
Comment (duanmengkk) [inline hack/globalnode_patch.sh]: There is no need to submit this code segment.

File: pkg/kubenest/controlplane/coredns/coredns.go [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+	if err != nil {
+		return err
+	}
+	//create core-dns rbac
+	err = EnsureCoreDnsRBAC(client, namespace, name)
+	if err != nil {
+		return err
+	}
+	//service in host has been created in the NewVirtualClusterServiceTask
+	//create core-dns deployment
```
Comment (duanmengkk) [inline pkg/kubenest/controlplane/coredns/coredns.go]: delete meaningless comment

File: pkg/kubenest/controlplane/coredns/coredns.go [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+	"k8s.io/client-go/dynamic"
+	clientset "k8s.io/client-go/kubernetes"
+
+	"github.com/kosmos.io/kosmos/pkg/kubenest/manifest/controlplane/coredns/host"
+	"github.com/kosmos.io/kosmos/pkg/kubenest/manifest/controlplane/coredns/virtualcluster"
+	"github.com/kosmos.io/kosmos/pkg/kubenest/util"
+)
+
+func EnsureHostCoreDns(client clientset.Interface, name, namespace string) error {
+	//create core-dns cm for host
```
Comment (duanmengkk) [inline pkg/kubenest/controlplane/coredns/coredns.go]: delete meaningless comment

File: pkg/kubenest/controlplane/coredns/coredns.go [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+	//create core-dns deployment
+	err = installCoreDnsDeployment(client, name, namespace)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func EnsureVirtualClusterCoreDns(dynamicClient dynamic.Interface, templateMapping map[string]interface{}) error {
+	// create core-dns endpoints for virtualcluster
```
Comment (duanmengkk) [inline pkg/kubenest/controlplane/coredns/coredns.go]: delete meaningless comment

File: pkg/kubenest/controlplane/coredns/coredns.go [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+	err := installCoreDnsEndpointsInVirtualCluster(dynamicClient, templateMapping)
+	if err != nil {
+		return err
+	}
+	// create core-dns service in virtualcluster
```
Comment (duanmengkk) [inline pkg/kubenest/controlplane/coredns/coredns.go]: **delete meaningless comment**

File: pkg/kubenest/manifest/controlplane/coredns/host/manifest_deployment.go [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+                  operator: In
+                  values:
+                  - kube-dns
+              topologyKey: kubernetes.io/hostname
+            weight: 100
+      containers:
+      - args:
+        - -conf
+        - /etc/coredns/Corefile
+        image: {{ .ImageRepository }}/coredns:v1.9.3
```
Comment (duanmengkk) [inline pkg/kubenest/manifest/controlplane/coredns/host/manifest_deployment.go]:
```suggestion
        image: {{ .ImageRepository }}/coredns: {{ .CoreDNSImageTag }}
```

File: pkg/kubenest/manifest/controlplane/coredns/host/manifest_configmap.go [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+        kubernetes cluster.local in-addr.arpa ip6.arpa {
+           pods insecure
+           fallthrough in-addr.arpa ip6.arpa
+           ttl 30
+           kubeconfig /etc/apiserver/kubeconfig
+        }
+        prometheus :9153
+        forward . /etc/resolv.conf {
+           max_concurrent 1000
+        }
```
Comment (OrangeBao) [inline pkg/kubenest/manifest/controlplane/coredns/host/manifest_configmap.go:23]: Can be configured here

**Pull request merged**